maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -157,9 +157,9 @@ class IOSDriver(
 
     fun viewHierarchy(): TreeNode {
         LOGGER.info("Requesting view hierarchy of the screen")
-        val hierarchyResult = iosDevice.viewHierarchy().get()
-        LOGGER.info("Depth of the screen is ${hierarchyResult?.depth ?: 0}")
-        if (hierarchyResult?.depth != null && hierarchyResult.depth > WARNING_MAX_DEPTH) {
+        val hierarchyResult = iosDevice.viewHierarchy().expect {  }
```

Comment:
Actually that is the problem, right now the cases it errors specifically where the XCUITest server returns 5xx or 4xx it basically gives null, we should be failing this hard since we don't expect any known errors to happen here. Let me know if you think of any error that should be handled here?

Reply:
Gotcha, no worries then. Thanks for the context Amanjeet